pkg/resources/cruisecontrol/deployment.go

```diff
@@ -174,8 +174,8 @@ fi`},
 	}
 }
 
-func generatePodAnnotations(kafkaCluster *v1beta1.KafkaCluster, log logr.Logger) map[string]string {
-	hashedCruiseControlCapacityJson := sha256.Sum256([]byte(GenerateCapacityConfig(kafkaCluster, log)))
+func generatePodAnnotations(kafkaCluster *v1beta1.KafkaCluster, log logr.Logger, config *corev1.ConfigMap) map[string]string {
+	hashedCruiseControlCapacityJson := sha256.Sum256([]byte(GenerateCapacityConfig(kafkaCluster, log, config)))
```

Comment:
Inject capacityConfig as parameter instead of invoking again `GenerateCapacityConfig`